erpnext/controllers/accounts_controller.py

```diff
@@ -479,7 +479,19 @@ def set_advances(self):
 			if d.against_order:
 				allocated_amount = flt(d.amount)
 			else:
-				amount = self.rounded_total or self.grand_total
+				party_account_currency = self.get('party_account_currency')
+
+				if not party_account_currency:
+					party_type, party = self.get_party()
+
+					if party_type and party:
+						party_account_currency = get_party_account_currency(party_type, party, self.company)
```

Comment:
this is not required for set_advances method. Because the function is only for SI and PI and in both doctype party_account_currency field exists.